### ISSUE TITLE
Tab does not work as a trigger character for 

### ISSUE THREAD

**Issue by ericbiewener** [microsoft/vscode#89826]
- VSCode Version: 1.41.1
- OS Version: Mac 10.14.6

Steps to Reproduce:

Register a CompletionProvider with tab (`\t`) as a trigger character:

```js
vscode.languages.registerCompletionItemProvider(
  'yaml', myCompletionItemProvider, ' ', '\n', '\t'
)
```
In the above example, the space and new line characters successfully trigger my completion provider, but not tab. This is the case whether my text editor is using tabs or spaces.

**Comment by jrieken** [comment]
Yeah, that kinda wanted because using space, newline, and tab is a bit of an anti-usage of trigger characters. What is you use-case for this? What language needs such trigger characters?

**Comment by ericbiewener** [comment]
My use case is autocompletion for yaml. Since yaml is whitespace significant, the autocompletion needs to suggest different options depending on indentation levels.

**Comment by jrieken** [comment]
> Since yaml is whitespace significant, the autocompletion needs to suggest different options depending on indentation levels.

That I understand, but all the whitespace and indentation levels are inside the document that you receive when being triggered (in whatever way). Tho, I do wonder if yaml users expect that IntelliSense is triggered by whitespace... Our intent for them is to trigger exclusive completions in cases like member completion, e.g `.` being a trigger character. "General" completion is trigger by typing words (quick suggestion) and by explicitly requesting completions (ctrl+space)

**Comment by ericbiewener** [comment]
I guess it's a question of whether VS Code should block tab as a trigger character because it isn't desirable in the vast majority of use cases, _even if there may be some rare cases where it would be useful_. I'd advocate for VS Code leaving that decision up to extension authors rather than simply preventing the option outright.

**Comment by jrieken** [comment]
> I guess it's a question of whether VS Code should block tab as a trigger character because it isn't desirable in the vast majority of use cases

It's not that we are blocking anything but tab doesn't reach us like other characters do. To implement trigger characters we have a "typing listener" which spies on what's being typed. Pressing tab, doesn't perform a type-operation but a more complex insert operation because pressing tab might mean "insert 4 spaces", "insert 2 spaces", "insert \t" etc. That's why we are bling on that eye. 

Rephrasing my question, what is a good sample yaml-snippet in which tab should trigger completions (from only your provider)?

**Comment by ericbiewener** [comment]
```yaml
- someObject:
    |
```

In the above example, imagine I have just tabbed to the pipe character.

**Comment by Brianzchen** [comment]
@jrieken Does my use case with flowtype above ☝️ give reason to implement this change?

**Comment by rikinsk** [comment]
Ran into the same issue. We're trying to achieve completions for yaml files when tab is hit
